File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
import java.util.LinkedList;
import java.util.List;
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        List<Integer> list = new LinkedList<>(); 
        while(list1 != null){
            list.add(list1.val);
            list1 = list1.next;
        }
        while(list2 != null){
            list.add(list2.val);
            list2 = list2.next;
        }
        var l = list.stream().sorted((v1,v2) -> v2.compareTo(v1)).toList();
        ListNode out = null;
        for(int i = 0; i < l.size(); i++){
            var val = l.get(i);
            out = new ListNode(val, out);
        }
        return out;
    }
}

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        if list1 == None and list2 != None:
            return list2
        if list2 == None and list1 != None:
            return list1
        if list1 == None and list2 == None:
            return None
        
        head = None
        if list1.val <= list2.val:
            head = ListNode(list1.val, head)
            list1 = list1.next
        else:
            head = ListNode(list2.val, head)
            list2 = list2.next
        
        while list1 and list2:
            if list1.val <= list2.val:
                head = ListNode(list1.val, head)
                list1 = list1.next
            else:
                head = ListNode(list2.val, head)
                list2 = list2.next
        
        while list1:
            head = ListNode(list1.val, head)
            list1 = list1.next
        
        while list2:
            head = ListNode(list2.val, head)
            list2 = list2.next
        
        nwList = None
        while head:
            nwList = ListNode(head.val, nwList)
            head = head.next
        
        
        return nwList